src/daemon/ca/manager.rs

```diff
@@ -807,17 +807,33 @@ impl CaManager {
     /// Note: this function can be called manually through the API, but is normally
     ///       triggered in the background, every 10 mins by default, or as configured
     ///       by 'ca_refresh' in the configuration.
-    pub async fn cas_refresh_all(&self, actor: &Actor) {
+    pub async fn cas_refresh_all(&self, started: Timestamp, actor: &Actor) {
         if let Ok(cas) = self.ca_store.list() {
             let mut updates = vec![];
 
+            // Set threshold hours if it was configured AND this server has been started
+            // longer ago than the hours specified. Otherwise we risk that *all* children
+            // without prior recorded status are suspended on upgrade, or that *all* children
+            // are suspended if the server had been down for more than the threshold hours.
+            let threshold_hours = self
+                .config
+                .suspend_child_after_inactive_hours
+                .map(|hours| {
```

Comment:
Couldn't this be more simply done using [filter](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.filter) ?

Reply:
Should we log that we are delaying the check?

Reply:
Now using filter, thanks for the pointer.. I don't think we need to log. It would be incompatible with filter again, but more importantly I like to keep logging quiet. There is no problem in postponing.